src/parquet/file/writer-internal.h

```diff
@@ -67,12 +67,14 @@ class SerializedPageWriter : public PageWriter {
 class RowGroupSerializer : public RowGroupWriter::Contents {
  public:
   RowGroupSerializer(int64_t num_rows, const SchemaDescriptor* schema, OutputStream* sink,
-      format::RowGroup* metadata, MemoryAllocator* allocator)
+      format::RowGroup* metadata, MemoryAllocator* allocator,
+      const std::shared_ptr<WriterProperties>& properties)
```

Comment:
Passing a reference should be enough, the ParquetFileWriter already owns the properties and this object, thus it has a longer lifetime.